x/datadeal/keeper/reward.go

```diff
@@ -0,0 +1,58 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	distrtypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	"github.com/medibloc/panacea-core/v2/types/assets"
+	oracletypes "github.com/medibloc/panacea-core/v2/x/oracle/types"
+)
+
+// DistributeRewards distributes reward to oracles for data verification and delivery
+func (k Keeper) DistributeRewards(ctx sdk.Context, dealID uint64, oracles map[string]*oracletypes.OracleValidatorInfo) {
+	deal, err := k.GetDeal(ctx, dealID)
+	if err != nil {
+		panic(err)
+	}
+
+	dealAddress, err := sdk.AccAddressFromBech32(deal.GetAddress())
+	if err != nil {
+		panic(err)
+	}
+
+	// calculate oracle commission
+	totalBudget := deal.GetBudget().Amount.ToDec()
+	maxNumData := sdk.NewIntFromUint64(deal.GetMaxNumData()).ToDec()
+	oracleCommission := k.oracleKeeper.GetParams(ctx).OracleCommissionRate
+	pricePerData := sdk.NewCoin(assets.MicroMedDenom, totalBudget.Quo(maxNumData).TruncateDec().Mul(oracleCommission).TruncateInt())
```

Comment:
Thanks for your work! 👍  But, I have one question. Currently, consideration of the tx fee is not included in the calculation process, is this not necessary? I wonder if it is an automated process or if the fee itself is not used.

Reply:
Yes. AFAIK, it is consumed automatically whenever there is a read or write to the store. 
You can refer the docs [here](https://docs.cosmos.network/v0.46/basics/gas-fees.html#main-gas-meter) :)